### ISSUE TITLE
Failure to update to 0.10.0 - compile error

### ISSUE THREAD

**Issue by ghstridr** [JunSuzukiJapan/atom-keyboard-macros#23]
Atom v1.18.0 x64 on Windows 10.
Updating failed. Uninstall and reinstall of package also fails.
This is what it showed me after trying to update three times:
> keyboard-layout@2.0.13 install C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\node_modules\atom-keyboard-macros\node_modules\keyboard-layout
> node-gyp rebuild

C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\node_modules\atom-keyboard-macros\node_modules\keyboard-layout>if not defined npm_config_node_gyp (node "C:\Users\ghstridr\AppData\Local\atom\app-1.18.0\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\ghstridr\AppData\Local\atom\app-1.18.0\resources\app\apm\bin\\..\node_modules\node-gyp\bin\node-gyp.js" rebuild ) 
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\node_modules\atom-keyboard-macros\node_modules\keyboard-layout\build\binding.sln]
C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29
`-- (empty)

gyp ERR! build error 
gyp ERR! stack Error: `C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\ghstridr\AppData\Local\atom\app-1.18.0\resources\app\apm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Windows_NT 10.0.16251
gyp ERR! command "C:\\Users\\ghstridr\\AppData\\Local\\atom\\app-1.18.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\ghstridr\\AppData\\Local\\atom\\app-1.18.0\\resources\\app\\apm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\node_modules\atom-keyboard-macros\node_modules\keyboard-layout
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\package.json'
npm WARN apm-install-dir-117631-11488-vtr7ym.3e1z7low29 No description
npm WARN apm-install-dir-117631-11488-vtr7ym.3e1z7low29 No repository field.
npm WARN apm-install-dir-117631-11488-vtr7ym.3e1z7low29 No README data
npm WARN apm-install-dir-117631-11488-vtr7ym.3e1z7low29 No license field.
npm ERR! Windows_NT 10.0.16251
npm ERR! argv "C:\\Users\\ghstridr\\AppData\\Local\\atom\\app-1.18.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\ghstridr\\AppData\\Local\\atom\\app-1.18.0\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\ghstridr\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\ghstridr\\.atom\\.apmrc" "install" "C:\\Users\\ghstridr\\AppData\\Local\\Temp\\d-117631-11488-kifq50.orl8281tt9\\package.tgz" "--runtime=electron" "--target=1.3.15" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! keyboard-layout@2.0.13 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the keyboard-layout@2.0.13 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the keyboard-layout package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs keyboard-layout
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls keyboard-layout
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\ghstridr\AppData\Local\Temp\apm-install-dir-117631-11488-vtr7ym.3e1z7low29\npm-debug.log
npm ERR! code 1

**Comment by kankaristo** [comment]
I'm also getting this with Atom 1.19.0-beta7 on Fedora 26.

Slightly different output:
```
Installing “atom-keyboard-macros@0.10.0” failed.
/bin/bash: scl: line 1: syntax error: unexpected end of file
/bin/bash: error importing function definition for `scl'
/bin/bash: module: line 1: syntax error: unexpected end of file
/bin/bash: error importing function definition for `module'


> keyboard-layout@2.0.13 install /tmp/apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9/node_modules/atom-keyboard-macros/node_modules/keyboard-layout
> node-gyp rebuild

/tmp/apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9
└── (empty)


sh: scl: line 1: syntax error: unexpected end of file
sh: error importing function definition for `scl'
sh: module: line 1: syntax error: unexpected end of file
sh: error importing function definition for `module'
sh: scl: line 1: syntax error: unexpected end of file
sh: error importing function definition for `scl'
sh: module: line 1: syntax error: unexpected end of file
sh: error importing function definition for `module'
/bin/bash: scl: line 1: syntax error: unexpected end of file
/bin/bash: error importing function definition for `scl'
/bin/bash: module: line 1: syntax error: unexpected end of file
/bin/bash: error importing function definition for `module'
/bin/sh: scl: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `scl'
/bin/sh: module: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `module'
gyp: Call to 'node -e "require('nan')"' returned exit status 0 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/atom-beta/resources/app/apm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 4.11.11-300.fc26.x86_64
gyp ERR! command "/usr/share/atom-beta/resources/app/apm/bin/node" "/usr/share/atom-beta/resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /tmp/apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9/node_modules/atom-keyboard-macros/node_modules/keyboard-layout
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/tmp/apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9/package.json'
npm WARN apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9 No description
npm WARN apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9 No repository field.
npm WARN apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9 No README data
npm WARN apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9 No license field.
npm ERR! Linux 4.11.11-300.fc26.x86_64
npm ERR! argv "/usr/share/atom-beta/resources/app/apm/bin/node" "/usr/share/atom-beta/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/home/sami/.atom/.apm/.apmrc" "--userconfig" "/home/sami/.atom/.apmrc" "install" "/tmp/d-11777-27497-it0j4z.zhl7hkt9/package.tgz" "--runtime=electron" "--target=1.6.9" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! keyboard-layout@2.0.13 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the keyboard-layout@2.0.13 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the keyboard-layout package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs keyboard-layout
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls keyboard-layout
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /tmp/apm-install-dir-11777-27497-d1lg8q.sqz4uqh0k9/npm-debug.log
npm ERR! code 1
```

**Comment by audreytoskin** [comment]
This compile error is preventing me from installing in the first place :(

```
↪  atom --version
Atom    : 1.23.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

* Fedora 26 Workstation x86_64
* Atom 1.23.1 (installed via the official RPM release on GitHub)
* system NodeJS 8.9.1 (installed via standard Fedora repos)
* NPM 5.5.1

**Comment by audreytoskin** [comment]
On Linux, I noticed that installing the development packages for libxkbfile solved this installation error. On Debian-based systems, you'd want to install `libxkbfile-dev`; on Red Hat-based systems, the package is `libxkbfile-devel`.

On Mac, @haberdashPI noted says "Anaconda is installed in my PATH and its version of c++ was overriding xcode's version. I removed it from my path and installed this from the command line successfully.

See #26
